File: oop/code/python_code/oop/inheritance/Student.py
from oop.inheritance.User import *
from oop.inheritance.StudentStatus import *

class Student(User):

	__status = StudentStatus.ACTIVE

	def __init__(self, name: str, email: str, batch_name: str, psp: int) -> None:
		super().__init__(name, email)
		self.__batch_name = batch_name
		
		if (psp < 0 or psp > 100):
			raise Exception("PSP should be between 0 and 100")
		self.__psp = psp

	def print_details(self) -> None:
		print("In Student", self.get_name(), self.__batch_name)

	def change_batch(self, batch_name: str) -> None:
		self.batch_name = batch_name

File: oop/code/java/src/main/java/com/scaler/lld/scaler/Student.java
package com.scaler.lld.scaler;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Student extends User {

    private String batchName;
    private Integer psp;
    private StudentStatus status = StudentStatus.ACTIVE; // ACTIVE, PAUSED, COMPLETED

    // Define parametrised constructor

    public Student(String name, String email, String batchName, Integer psp) {
        super(name, email);
        this.batchName = batchName;

        if (psp < 0 || psp > 100) {
            throw new IllegalArgumentException("PSP should be between 0 and 100");
        }
        this.psp = psp;
    }

    public Student() {
    }

    void changeBatch(String batchName) {
        this.batchName = batchName;
    }

    @Override
    public void printInfo() {
        System.out.println("\nStudent: " + getName() + " " + getBatchName());
    }

}